pkgs/os-specific/linux/kernel/perf/perf-dmesg-path.diff

```diff
@@ -0,0 +1,13 @@
+diff --git tools/perf/util/evsel.c tools/perf/util/evsel.c
+index a8a5ff87c..aa602aae6 100644
+--- tools/perf/util/evsel.c
++++ tools/perf/util/evsel.c
+@@ -3053,7 +3053,7 @@ int evsel__open_strerror(struct evsel *evsel, struct target *target,
+ 
+ 	return scnprintf(msg, size,
+ 	"The sys_perf_event_open() syscall returned with %d (%s) for event (%s).\n"
+-	"/bin/dmesg | grep -i perf may provide additional information.\n",
++	"dmesg | grep -i perf may provide additional information.\n",
```

Comment:
could be a substituteInPlace instead

Reply:
I initially wanted to upstream this but contributing to the kernel is not just opening a PR somewhere, so I didn't do that.

Also I don't really want this to silently ignore this in the future if the code is changed, hence I used a patch.

Reply:
```suggestion
+	"\"dmesg | grep -i perf\" may provide additional information.\n",
```
like this?

Reply:
OK I tripped over the message in a way I didn't expect. reading this I assumed `perf may provide ...` is the string one should look for, but apparently it isn't.

So while my initial suggestion is invalid (edited accordingly), your suggestion is reasonable (and should probably upstreamed) :+1:

Reply:
I am not sure if the kerne list likes my mail provider nor do I have the time to dig into how to setup git-mail or motivation to subscribe to a mailing list to send this biennial change.

I already asked a friend if they can upstream the first version of this patch and it took them way to long to send out the mail despite having it done before which definitely didn't spark motivation in me.

Reply:
fwiw for small providers (e.g. my personal mailserver) mails get usually graylisted when sent to lore.kernel.org and as a result it'll take about 30mins until it's there, so you need a bit of patience at least.

If noone else steps up, I can also try to upstream it (last time I threw a patch there was not so long ago), but first things first, let's get this into nixpkgs. For that, would you mind applying your suggestion? Then we'd have the version we'd also intend to upstream in here which means that (1) anybody can theoretically upstream it easily and (2) as soon as it's upstreamed, we'll get a clearer "reversed patch" error from patch rather than a "patch does not apply" error which makes it clear that it's sufficient to just drop the patch from our package definition.

Reply:
That sounds like a plan! I applied my suggestion and rebased.